t/27-consoles-vnc.t

```diff
@@ -50,4 +52,23 @@ subtest 'handling VNC stall, malformed RFB protocol on re-connect' => sub {
     is scalar @sent, 1, 'no further message sent' or diag explain \@sent;
 };
 
+subtest 'handling connect timeout' => sub {
+    $bmwqemu::vars{VNC_CONNECT_TIMEOUT_LOCAL} = 5;
+    $bmwqemu::vars{VNC_CONNECT_TIMEOUT_REMOTE} = 10;
+    my $attempts = 0;
+    $c->hostname('127.0.0.100');
+    $inet_mock->redefine(new => sub { ++$attempts; undef });
+    _setup_rfb_magic;
+    combined_like {
+        throws_ok { $c->login } 'OpenQA::Exception::VNCSetupError', 'dies on connect timeout' }
+    qr/.*Error connecting to.*/, 'error logged';
+    is $attempts, 7, 'login attempts for local hostname';
+    $attempts = 0;
+    $c->hostname('10.161.145.95');
+    combined_like {
+        throws_ok { $c->login } 'OpenQA::Exception::VNCSetupError', 'dies on connect timeout (2)'
+    } qr/.*Error connecting to.*/, 'error logged (2)';
```

Comment:
Is this trying to actually connect to this adress? I would prefer to mock the real connection attempt or at least use a non-reachable localhost address. What about 127.0.0.2?

Reply:
ok, cool